src/http/client.rs

```diff
@@ -76,6 +137,25 @@ impl Client {
     }
 }
 
+/// Returned from [`Client::start_request`], this represents a handle to a
+/// response which has not arrived yet. Call [`FutureIncomingResponse::get`]
+/// to wait for the response.
+pub struct FutureIncomingResponse(WasiFutureIncomingResponse);
+
+impl FutureIncomingResponse {
```

Comment:
I think this can be https://doc.rust-lang.org/std/future/trait.IntoFuture.html ?

Reply:
I've now updated it to use `impl Future`.